### PR TITLE
Pre-allocate rclcpp::Time's in the main loop

### PR DESCRIPTION
**Pull request by AndyZe**
This is probably a negligible performance improvement but I guess it doesn't hurt.

### CODE REVIEW

**Review by aprotyas — COMMENTED** [review]
No strong opinion, but I think the code was more readable before this change, since `begin_last` and `end` are polluting the outer scope now. Compilers are probably good with optimizing these in-loop definitions too.

**Comment by AndyZe** [comment]
Well, I'm not big on trusting compilers so I dropped some benchmarking code in. Basically it adds up the duration of the loop internals minus the sleep() for 10k iterations (attached). I did 10 runs with the status quo vs 10 runs with this PR, ABAB style.

Basically there was no difference. However, I still think we should merge the PR because this is one of the most critical bits of code and it feels weird to depend on compiler optimization for this critical bit.

[benchmark.txt](https://github.com/ros-controls/ros2_control/files/8049381/benchmark.txt)

**Review by aprotyas — APPROVED** [review]
> Basically there was no difference. However, I still think we should merge the PR because this is one of the most critical bits of code and it feels weird to depend on compiler optimization for this critical bit.

Hmm, I think there won't be any maintainability concerns either from moving said variables to the outer scope given how small this file is (and probably will remain). I don't see a strong reason to oppose this then. LGTM

File: controller_manager/src/ros2_control_node.cpp [hunk 1]
```diff
@@ -41,17 +41,19 @@ int main(int argc, char ** argv)
     RCLCPP_INFO(cm->get_logger(), "update rate is %d Hz", cm->get_update_rate());
 
     rclcpp::Time begin = cm->now();
+    rclcpp::Time begin_last(begin);
+    rclcpp::Time end(begin);
```
Comment (destogl) [inline controller_manager/src/ros2_control_node.cpp:45]: Some nit-picking on a nitpick :smile: 

```suggestion
    rclcpp::Time begin_last;
    rclcpp::Time end;
```

**Review by destogl — APPROVED** [review]
Thanks for this nitpick :)

@AndyZe and @aprotyas I would also like to hear your opinion on #645.

File: controller_manager/src/ros2_control_node.cpp (hunk 1 above)
Comment (AndyZe) [reply controller_manager/src/ros2_control_node.cpp:45]: I think we should initialize them, though. I could use this instead, if you prefer:

```
rclcpp::Time begin_last = begin;
rclcpp::Time end = begin;
```

**Comment by destogl** [comment]
closing in favor of #646

**Pull request closed without merging**